File: examples/run-code-in-a-sandbox/verify.mjs
/**
 * Proves this example works: boot a sandbox, exec python3, assert "42", dispose.
 * Token/project from env (CI) or ~/.config/tenki/config.yaml (local `tenki login`).
 * Exits non-zero on any failure.
 */
import { TenkiSandbox, stdoutText } from "@tenkicloud/sandbox";
import { readFileSync } from "node:fs";
import { homedir } from "node:os";

const cfg = (key) => {
	try {
		const c = readFileSync(`${homedir()}/.config/tenki/config.yaml`, "utf8");
		return (c.match(new RegExp(`^${key}:\\s*(.+)$`, "m"))?.[1] ?? "").trim();
	} catch {
		return "";
	}
};

const authToken = process.env.TENKI_AUTH_TOKEN || process.env.TENKI_API_KEY || cfg("auth_token");
const projectId = process.env.TENKI_PROJECT_ID || cfg("current_project_id");
const workspaceId = process.env.TENKI_WORKSPACE_ID || cfg("current_workspace_id");
if (!authToken) {
	console.error("No token. Set TENKI_AUTH_TOKEN, or run `tenki login`.");
	process.exit(1);
}

const tenki = new TenkiSandbox({ authToken });
let sandbox;
try {
	sandbox = await tenki.createAndWait({ cpuCores: 1, memoryMb: 1024, projectId, workspaceId });
	const r = await sandbox.exec("python3", { args: ["-c", "print(6 * 7)"] });
	if (!(r.exitCode === 0 && stdoutText(r).trim() === "42")) {
		throw new Error(`unexpected result: exit ${r.exitCode}, stdout ${JSON.stringify(stdoutText(r))}`);
	}
	console.log("✓ run-code-in-a-sandbox: create → exec python3 → 42 → dispose");
} catch (e) {
	console.error("✗ " + (e?.message ?? e));
	process.exitCode = 1;
} finally {
	if (sandbox) {
		try {
			await sandbox[Symbol.asyncDispose]();
		} catch {
			/* self-reaps via idle/lifetime caps */
		}
	}
}

File: examples/run-code-python/verify.py
"""
Proves this example works against live Tenki: boot a sandbox, exec python3,
assert "42", dispose. No LLM needed — this is the whole example, asserted.

Needs Python 3.10+ and the dep in requirements.txt. Token/project from env (CI)
or ~/.config/tenki/config.yaml (local `tenki login`). Exits non-zero on failure.
"""
import os
import sys

from tenki_sandbox import Sandbox


def cfg(key):
    try:
        with open(os.path.expanduser("~/.config/tenki/config.yaml")) as f:
            for line in f:
                if line.startswith(key + ":"):
                    return line.split(":", 1)[1].strip()
    except FileNotFoundError:
        pass
    return ""


token = os.environ.get("TENKI_AUTH_TOKEN") or os.environ.get("TENKI_API_KEY") or cfg("auth_token")
if not token:
    print("No token. Set TENKI_AUTH_TOKEN, or run `tenki login`.")
    sys.exit(1)

# Python SDK auth gap: a bare `tenki login` session token must be sent as a
# cookie (prefix `cookie:`); a `tk_` API key works as-is. (Node SDK auto-detects.)
if not token.startswith(("tk_", "ory_st_", "cookie:")):
    token = f"cookie:{token}"

opts = {"auth_token": token, "cpu_cores": 1, "memory_mb": 1024}
project_id = os.environ.get("TENKI_PROJECT_ID") or cfg("current_project_id")
workspace_id = os.environ.get("TENKI_WORKSPACE_ID") or cfg("current_workspace_id")
if project_id:
    opts["project_id"] = project_id
if workspace_id:
    opts["workspace_id"] = workspace_id

try:
    # create -> exec -> assert -> dispose (the `with` block terminates the VM).
    with Sandbox.create(**opts) as sb:
        r = sb.exec("python3", "-c", "print(6 * 7)")
        if not (r.exit_code == 0 and r.stdout_text.strip() == "42"):
            raise AssertionError(
                f"unexpected result: exit {r.exit_code}, stdout {r.stdout_text!r}, stderr {r.stderr_text!r}"
            )
    print("✓ run-code-python: create → exec python3 → 42 → dispose")
except Exception as e:  # noqa: BLE001
    print(f"✗ {type(e).__name__}: {e}")
    sys.exit(1)
